<launch>
        <!-- <arg name="file_name" value="carla_parking"/> -->
        <arg name="file_name" value="simul_test"/>
        <!-- <arg name="file_name" value="parkinglot_parking"/> -->
        <arg name="carla" value="true"/>
                
        <!-- pose parser -->
        <include file="$(find mm_pose_parser)/launch/mm_pose_parser.launch">
                <arg name="file_name" value="$(arg file_name)"/>
                <arg name="carla" value="$(arg carla)"/>
        </include>

        <!-- tf -->
        <include file="$(find mm_tf2_broadcaster)/launch/tf2_broadcast.launch"/>
        
        <!-- vis -->
        <include file="$(find mm_vis)/launch/mm_vis.launch"/>

        <!-- udp -->
        <include file="$(find mm_udp)/launch/udp_bridge.launch"/>

        <!-- ogm -->
        <include file="$(find mm_ogm)/launch/mm_ogm.launch"/>

        <!-- lane -->
        <include file="$(find yolopv2_simul)/launch/lane_detector.launch"/>

        <!-- traffic -->
        <node pkg="yolopv2_simul" name="traffic_detector" type="traffic.py" output="log"/>

        <!-- main planner -->
        <node pkg="mm_planner" name="mm_planner" type="mm_planner" output="screen">

                <!--subscribe-->
                <remap from="~pose" to="/mm_pose_parser/pose"/>
                <remap from="~ogm" to="/mm_ogm/occupancy_grid"/>
                <!--publish-->
                <remap from="~command" to="/ctrl_cmd"/>

                <param name="path" value="/home/heven/catkin_ws/src/heven_simul_2025/mm_common/csv/main.csv"/>
                <param name="mission_points_path" value="/home/heven/catkin_ws/src/heven_simul_2025/mm_planner/config/simul_missionpoints.csv"/>

                <!-- Mission parameters -->
                <param name="dist_threshold" value="2.0"/>
                <param name="traffic_dist_threshold" value="5.0"/>
                <param name="slow_dist_threshold" value="20.0"/>
                <param name="current_mission" value="0"/>
                <param name="mission_on" value="false"/>

                <!-- Stanley parameters -->
                <param name="lad" value="3.0"/>
                <param name="vel" value="30.0"/>
                <param name="stanley_k" value="2.0"/>
                <param name="pid_kp" value="0.3"/>
                <param name="pid_ki" value="0.0"/>
                <param name="pid_kd" value="0.01"/>

                <!-- DWA parameters -->
                <param name="dwa_lad" value="15.0"/>
                <param name="dwa_dt" value="1.5"/>
                <param name="dwa_window" value="15.0"/>
                <param name="dwa_goal_weight" value="200.0"/>
                <param name="dwa_obs_weight" value="600.0"/>
                <param name="dwa_min_speed" value="17.0"/>
                <param name="dwa_max_speed" value="22.0"/>
                <param name="dwa_min_omega" value="-55.0"/>
                <param name="dwa_max_omega" value="55.0"/>
        </node>
</launch>
